clamav/util.py

```diff
@@ -70,32 +71,43 @@ def scan_stream(fileobj):
     return status, signature_or_none
 
 
-def scan_container_image(image_reference: str):
-    logger.debug(f'scanning container image {image_reference}')
-
+def iter_image_files(
+    container_image_reference: str,
+) -> typing.Tuple[typing.IO, str]:
```

Comment:
it is not _one_ tuple, but an iterable yielding such-typed tuples. if adding type hints, please be precise. otherwise, they are misleading and harm more than help.